x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/Match.java

```diff
@@ -113,4 +119,21 @@ protected TypeResolutions.ParamOrdinal queryParamOrdinal() {
     public Expression field() {
         return field;
     }
+
+    @Override
+    public String functionType() {
+        return isOperator() ? "operator" : super.functionType();
+    }
+
+    @Override
+    public String functionName() {
+        return isOperator() ? ":" : super.functionName();
+    }
+
+    private boolean isOperator() {
+        if (isOperator == null) {
+            isOperator = source().text().toUpperCase(Locale.ROOT).startsWith(super.functionName()) == false;
```

Comment:
Shall we check the source text starts with `match(` and end with `)`, in case a field name starts with `match`?

Reply:
yes, that's a possibility. I've used a regex to address whitespaces between `match` and `(` and added some tests to check messages are returned correctly in 9ef72e7ce3e87e0e2e0c3dae5be0ffb5f2c6aa87